store/postgres/src/relational_queries.rs

```diff
@@ -932,7 +967,114 @@ impl<'a> QueryFilter<'a> {
+
+        let child_prefix = "i.";
+        let parent_prefix = "c.";
+
+        out.push_sql("exists (select 1");
+
+        out.push_sql(" from ");
+        out.push_sql(child_table.qualified_name.as_str());
+        out.push_sql(" as i");
+
+        out.push_sql(" where ");
+
+        // Join tables
+        if derivative.is_derived() {
+            // If the parent is derived,
+            // the child column is picked based on the provided attribute
+            // and the parent column is the primary key of the parent table
+            let child_column = child_table
+                .column_for_field(attribute)
+                .expect("Column for an attribute not found");
+            let parent_column = self.table.primary_key();
+
+            if child_column.is_list() {
+                // Type A: p.id = any(c.{parent_field})
+                out.push_sql(parent_prefix);
+                out.push_identifier(parent_column.name.as_str())?;
+                out.push_sql(" = any(");
+                out.push_sql(child_prefix);
+                out.push_identifier(child_column.name.as_str())?;
+                out.push_sql(")");
+            } else {
+                // Type B: p.id = c.{parent_field}
+                out.push_sql(parent_prefix);
+                out.push_identifier(parent_column.name.as_str())?;
+                out.push_sql(" = ");
+                out.push_sql(child_prefix);
+                out.push_identifier(child_column.name.as_str())?;
+            }
+        } else {
+            // If the parent is not derived, we do the opposite.
+            // The parent column is picked based on the provided attribute
+            // and the child column is the primary key of the child table
+            let parent_column = self
+                .table
+                .column_for_field(attribute)
+                .expect("Column for an attribute not found");
+            let child_column = child_table.primary_key();
+
+            if parent_column.is_list() {
+                // Type C: c.id = any(p.child_ids)
+                out.push_sql(child_prefix);
+                out.push_identifier(child_column.name.as_str())?;
+                out.push_sql(" = any(");
+                out.push_sql(parent_prefix);
+                out.push_identifier(parent_column.name.as_str())?;
+                out.push_sql(")");
+            } else {
+                // Type D: c.id = p.child_id
+                out.push_sql(child_prefix);
+                out.push_identifier(child_column.name.as_str())?;
+                out.push_sql(" = ");
+                out.push_sql(parent_prefix);
+                out.push_identifier(parent_column.name.as_str())?;
+            }
         }
+
+        out.push_sql(" and ");
+
+        // Match by block
+        out.push_sql(child_prefix);
+        out.push_identifier(BLOCK_RANGE_COLUMN)?;
+        out.push_sql(" @> ");
+        out.push_bind_param::<Integer, _>(&self.block)?;
```

Comment:
This needs to use `BlockRangeColumn::contains` - the way it is written now only works for mutable entities, but would fail for immutable ones since they don't have a `block_range` column, only a `block$` column

Reply:
f6721a3f344acdc80d0d5dfec1146854b95ee24f